components/constellation/constellation.rs

```diff
@@ -1722,7 +1760,7 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             TraversalDirection::Forward(delta) => {
                 for entry in self.joint_session_future(top_level_frame_id).take(delta) {
                     size = size + 1;
-                    table.insert(entry.frame_id, entry);
+                    table.insert(entry.frame_id, entry.clone());
```

Comment:
It's a pity we have to clone here, can we make the table have `&FrameState` entries?